08_project/alextmz/pkg/puppy/store/storer_test.go

```diff
@@ -0,0 +1,102 @@
+package store
+
+import (
+	"testing"
+
+	. "github.com/anz-bank/go-course/08_project/alextmz/pkg/puppy"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	p  []Puppy
+	st Storer
+}
+
+func (su *storerSuite) SetupTest() {
+	su.p = nil
+	var p = []Puppy{
+		{ID: 256, Breed: "Dogo", Colour: "White", Value: 500},
+		{ID: 512, Breed: "Mastiff", Colour: "Brindle", Value: 700},
+		{ID: 768, Breed: "Fila", Colour: "Golden", Value: 900},
+		{ID: 1024, Breed: "Wolfhound", Colour: "Gray", Value: -50},
+	}
+	su.p = append(su.p, p...)
+	err := su.st.CreatePuppy(&su.p[0])
+	su.Nil(err)
+}
+
+// TearDownTest cleans everything after each test runs.
+// It impacts test performance as things get re-populated soon afterwards,
+// but we're looking for reliable here not fast
+func (su *storerSuite) TearDownTest() {
+	for _, v := range su.p {
+		if _, err := su.st.ReadPuppy(v.ID); err == nil {
+			err := su.st.DeletePuppy(v.ID)
+			if err != nil {
```

Comment:
please consider
```go
if err := su.st.DeletePuppy(v.ID); err != nil {
    su.T()......
```

Reply:
Refactored - I don't think it applies anymore